File: suyigou_manager_web/src/main/java/com/suyigou/manager/controller/BrandController.java
package com.suyigou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.suyigou.pojo.TbBrand;
import com.suyigou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.ResultInfo;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    //查询所有品牌
    @RequestMapping("/findAll")
    public List<TbBrand> findAllBrand() {
        System.out.println(brandService);
        List<TbBrand> brandList = brandService.findAll();
        return brandList;
    }

    //分页查询
    @RequestMapping("/findByPage")
    public PageResult findByPage(@RequestParam(name = "curPage", required = true) int curPage,
                                 @RequestParam(name = "size", required = true) int size) {
        PageResult pageResult = brandService.findByPage(curPage, size);
        System.out.println("pageResult = " + pageResult);
        return pageResult;
    }

    //保存品牌
    @RequestMapping("/save")
    public ResultInfo save(@RequestBody TbBrand brand) {
        ResultInfo result = new ResultInfo();
        try {
            brandService.save(brand);
            result.setSuccess(true);
            result.setMsg("添加成功");
        } catch (Exception e) {
            result.setSuccess(false);
            result.setMsg("添加失败");
        }
        return result;
    }

    @RequestMapping("/findById")
    public TbBrand findById(@RequestParam(name = "id", required = true) Long id) {
        return brandService.findById(id);
    }

    //修改
    @RequestMapping("/update")
    public ResultInfo update(@RequestBody TbBrand brand) {
        ResultInfo result = new ResultInfo();
        try {
            brandService.update(brand);
            result.setSuccess(true);
            result.setMsg("更改成功");
        } catch (Exception e) {
            result.setSuccess(false);
            result.setMsg("更改失败");
        }
        return result;
    }

    //删除
    @RequestMapping("/delete")
    public ResultInfo delete(@RequestBody List<Long> ids) {
        ResultInfo result = new ResultInfo();
        try {
            brandService.delete(ids);
            result.setSuccess(true);
            result.setMsg("删除成功");
        } catch (Exception e) {
            result.setSuccess(false);
            result.setMsg("删除失败");
        }
        return result;
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand, int curPage, int size) {
        PageResult pageResult = brandService.findByPage(brand, curPage, size);
        return pageResult;
    }

}

File: suyigou_manager_web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope, $controller, brandService) {

    $controller('baseController', {$scope: $scope});

    //查找所有
    $scope.findAll = function () {
        brandService.findAll().success(function (result) {
            $scope.list = result.rows;
            $scope.paginationConf.totalItem = result.total;
        })
    };

    //分页查询
    $scope.findByPage = function (curPage, size) {
        brandService.findByPage(curPage, size).success(function (result) {
            $scope.list = result.rows;
            $scope.paginationConf.totalItem = result.total;
        });
    };

    //查找一个
    $scope.findById = function (id) {
        brandService.findById(id).success(function (result) {
            $scope.brand2save = result;
        });
    };

    //搜索
    $scope.search = function (curPage, size, entity) {
        brandService.search(curPage, size, entity)
            .success(function (result) {
                $scope.list = result.rows;
                $scope.paginationConf.totalItem = result.total;
            });
    };
    //保存
    $scope.save = function () {
        var res = null;
        if ($scope.brand2save.id == null) {
            //如果为空，则为添加
            res = brandService.add($scope.brand2save);
        } else {
            res = brandService.update($scope.brand2save);
        }
        res.success(function (result) {
                if (!result.success) {
                    alert(result.msg);
                }
                $scope.reloadList();
            }
        )
    };

    //删除
    $scope.delete = function () {
        brandService.delete($scope.delList).success(
            function (result) {
                if (!result.success) {
                    alert(result.msg);
                }
                $scope.delList = [];
                $scope.reloadList();
            }
        )
    };

})